static/js/composebox_typeahead.js

```diff
@@ -855,7 +855,15 @@ exports.content_typeahead_selected = function (item, event) {
         // Stream + topic mention typeahead; close the stream+topic mention syntax
         // with the topic and the final **.
         const start = -this.token.length;
-        beginning = beginning.slice(0, start) + item + "** ";
+        if (start !== 0) {
+            // We do not perform string slicing when the token
+            // length is 0 as it returns an empty string.
+            // The token length can be 0 for this case as we
+            // allow the setting up of the topic typeahead for
+            // completing the stream mention.
```

Comment:
I don't understand this comment; perhaps you can provide an example?

I think it'd be well worth writing a test for this case (it does seem like logic well worth verifying), so that might be the best way to provide such an example.

Reply:
5fc37c5f9b6af93ed2228215022f0e3c109fa688 is a useful reference; it does contain a `topic_list` test though it may not cover this block of code.

Reply:
I think this is the right track. The bug is that we are calling slice with (0,-0) which doesn't make sense.:

```
x = 'hello';
x.slice(0, -1); // 'hell'
x.slice(0, -2); // 'hel'
x.slice(0, -0); // ''
```

If Anders would like to avoid a conditional, we could go back to using `x.slice(0, x.length - start)` but a conditional plus comment should be fine here.